src/components/contact-us/LocationFields.jsx

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import FormField from '../FormField';
+import elementStyles from '../../styles/isomer-cms/Elements.module.scss';
+import _ from 'lodash';
+import { isEmpty } from '../../utils'
+
+const DEFAULT_NUM_OPERATING_FIELDS = 5;
+
+const LocationHoursFields = ({ 
+  operatingHours,
+  cardIndex,
+  onFieldChange,
+  errors,
+}) => {
+  
+  return (
+    <div className= "mt-4">
+      <h6> Operating Hours </h6>
+      { operatingHours && operatingHours.map( (operations, operationsIndex) => ( 
+        <div className= "mb-1" key={operationsIndex}>
+            <div className="d-flex flex-row">
+              <div className="flex-fill pr-1"> 
+                <FormField
+                  title="Days"
+                  id={`location-${cardIndex}-operating_hours-${operationsIndex}-days`}
+                  value={operations.days}
+                  onFieldChange={onFieldChange}
+                  errorMessage={errors[operationsIndex].days}
+                />
+              </div>
+              <div className="flex-fill pl-1">
+                <FormField
+                  title="Hours"
+                  id={`location-${cardIndex}-operating_hours-${operationsIndex}-time`}
+                  value={operations.time}
+                  onFieldChange={onFieldChange}
+                  errorMessage={errors[operationsIndex].time}
+                />
+              </div>
+            </div>
+            <div>
+              <FormField
+                title="Description"
+                id={`location-${cardIndex}-operating_hours-${operationsIndex}-description`}
+                value={operations.description}
+                onFieldChange={onFieldChange}
+                errorMessage={errors[operationsIndex].description}
+              />
+            </div>
+          <a className={elementStyles.formFixedText} id={`location-${cardIndex}-remove_operating_hours-${operationsIndex}`} href="#" onClick={onFieldChange}>
+            Remove
+          </a> 
+        </div>
+      ))} 
+      <div className = "mt-3">
+        { operatingHours.length < DEFAULT_NUM_OPERATING_FIELDS  
+          ? <a className={elementStyles.formLabel}  id={`location-${cardIndex}-add_operating_hours`} href="#" onClick={onFieldChange}>
+              Add operating hours
+            </a>
+          : <p className={elementStyles.formLabel}> Maximum 5 operating hours fields</p>
+        }
+      </div>
+  </div>
+  )
+};
+
+const LocationAddressFields = ({ 
+  title,
+  address,
+  cardIndex,
+  onFieldChange,
+  errors,
+}) => {
+
+  const [ errorMessage, setErrorMessage ] = useState('')
+
+  useEffect(() => { 
+    let newErrorMessage = '';
+    if (!isEmpty(errors)) {
+      if (errors.length === 1) {
+        newErrorMessage = errors[0]
+      } else {
+        errors.forEach((error, i) => {
+          newErrorMessage += error ? `Line ${i+1}: ${error} ` : ''
+        })
```

Comment:
> Left a few comments. In addition, there were some other things we could possibly consider modifying:
> 
> * Editing a field automatically shifts the editor view back to the start of a section (e.g. editing a location will bring you to the very first location), which is problematic for long sections - the viewer should only snap back to the item being edited
> * We might want to provide a way to add and delete sections - there's no way to remove or add a new section entirely currently
> * We could implement a check for any edited data and give a warning if the user is navigating away with unsaved changes, similar to editHomepage
> * One other thing is that on a site with no contact-us to begin with, there's no way to add a contact-us page. We could possible consider giving the option to create a new contact-us page, which gives default values? This could also be used in the case that the contact-us page is empty. Am open to suggestions on this

Things to work on:
1. Snap to end on creation https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView
2. Feedback section position in editor does not correlate with position in preview (add note on what it affects?)
3. Remove empty contacts array if no contacts.length (don't have a add/delete sections button)
4. Multiline address error is okay
5. Stylistic changes: 
- dangerouslySetHtml (how to style it consistently with <p>)
- View Map (padding)
- Online form
- Emails and phone numbers under contacts
6. Set background to white
7. Automatically format numbers

Things to check w designers:
1. Should links in general on preview pages work?
2. Feedback field position/ when you type how to correlate 
3. View Map url generation (how to tell users that we will generate the url automatically
4. How to deal with already erroneous data on mount (sanitise, validation) - also how to highlight which cards have errors?
5. Limit on # of contact cards/ location cards